cmd/shim/secret_scrub_test.go

```diff
@@ -133,24 +151,97 @@ func TestLoadSecretsToScrubFromFiles(t *testing.T) {
 	})
 }
 
-var (
-	//nolint:typecheck
-	//go:embed testdata/id_ed25519
-	sshSecretKey string
+func TestScrubSecretWrite(t *testing.T) {
+	t.Parallel()
+	envMap := map[string]string{
+		"secret1":      "secret1 value",
+		"secret2":      "secret2",
+		"sshSecretKey": sshSecretKey,
+		"sshPublicKey": sshPublicKey,
+	}
 
-	//nolint:typecheck
-	//go:embed testdata/id_ed25519.pub
-	sshPublicKey string
-)
+	env := []string{}
+	for k, v := range envMap {
+		env = append(env, fmt.Sprintf("%s=%s", k, v))
+	}
 
-func TestScrubSecretWrite(t *testing.T) {
-	secrets := []string{
+	envNames := []string{
 		"secret1",
-		"secret with space ",
-		sshSecretKey,
-		sshPublicKey,
+		"secret2",
+		"sshSecretKey",
+		"sshPublicKey",
+	}
+
+	secretToScrubInfo := core.SecretToScrubInfo{
+		Envs:  envNames,
+		Files: []string{},
 	}
-	s := "Not secret\nsecret1\nsecret with space\n" + sshSecretKey + "\n" + sshPublicKey
-	got := scrubSecretBytes(secrets, []byte(s))
-	require.Equal(t, "Not secret\n***\n***\n***\n***\n***\n***\n***\n***\n***\n\n***\n", string(got))
+
+	t.Run("multiline secret", func(t *testing.T) {
+		var buf bytes.Buffer
+		r, err := NewSecretScrubReader(&buf, "/", fstest.MapFS{}, env, secretToScrubInfo)
+		require.NoError(t, err)
+		input := "aaa\n" + sshSecretKey + "\nbbb\nccc"
```

Comment:
nit: would also be good to have a test like
```go
input := "aaa" + sshSecretKey + "bbb\nccc"
```

So we verify that the implementation doesn't rely on the secret being surrounded by `\n` too.